WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -566,30 +567,84 @@ public static void viewHistoryDetailForResult(Activity activity, Revision revisi
      * Load the post preview as an authenticated URL so stats aren't bumped
      */
     public static void browsePostOrPage(Context context, SiteModel site, PostModel post) {
-        if (site == null || post == null || TextUtils.isEmpty(post.getLink())) {
+        browsePostOrPageEx(context, site, post, RemotePreviewType.NOT_A_REMOTE_PREVIEW);
+    }
+
+    public static void previewPostOrPageForResult(
+            Activity activity,
+            SiteModel site,
+            PostModel post,
+            RemotePreviewType remotePreviewType
+    ) {
+        browsePostOrPageEx(activity, site, post, remotePreviewType);
+    }
+
+    private static void browsePostOrPageEx(
+            Context context,
+            SiteModel site,
+            PostModel post,
+            RemotePreviewType remotePreviewType) {
+        if (
+                site == null || post == null || TextUtils.isEmpty(post.getLink())
+                ||
+                (remotePreviewType == RemotePreviewType.REMOTE_PREVIEW_WITH_REMOTE_AUTO_SAVE
+                        && TextUtils.isEmpty(post.getAutoSavePreviewUrl()))
+        ) {
             return;
         }
 
         // always add the preview parameter to avoid bumping stats when viewing posts
-        String url = UrlUtils.appendUrlParameter(post.getLink(), "preview", "true");
+        String url =
```

Comment:
Do you mind moving this to a Utils method like `getPreviewUrlForPost`: 1. would be a bit easier to read and 2. it might be useful elsewhere.

Reply:
Yep, it makes sense. Done here cc3d05fc36d6f7e89aec9326032a929d1d2ca8fa